src/compiler/types/tests/stencil-types.spec.ts

```diff
@@ -0,0 +1,298 @@
+        };
+
+        const actualTypeName = updateTypeIdentifierNames(
+          typeReferences,
+          {},
+          stubComponentCompilerMeta().sourceFilePath,
+          expectedTypeName
+        );
+
+        expect(actualTypeName).toBe(expectedTypeName);
+      });
+
+      it("does not change a simple type when there's no import name to alias", () => {
+        const initialType = 'InitialType';
+        const basePath = '~/some/stubbed/path';
+        const typePath = `${basePath}/my-types`;
+
+        const componentCompilerMeta = stubComponentCompilerMeta({
+          sourceFilePath: `${basePath}/my-component.tsx`,
+        });
+        const typeReferences: d.ComponentCompilerTypeReferences = {
+          [initialType]: stubComponentCompilerTypeReference({ location: 'import', path: typePath }),
+        };
+        const typeImports = stubTypesImportData({
+          [typePath]: [
+            {
+              localName: 'SomeOtherType',
+            },
+          ],
+        });
+
+        const actualTypeName = updateTypeIdentifierNames(
+          typeReferences,
+          typeImports,
+          componentCompilerMeta.sourceFilePath,
+          initialType
+        );
+
+        expect(actualTypeName).toBe(initialType);
+      });
+    });
+
+    describe('path resolution', () => {
+      it('replaces a simple type for a relative path beginning with "."', () => {
+        const initialType = 'InitialType';
+        const expectedType = 'NonCollisionType';
+        const basePath = './some/stubbed/path';
+
+        testTypeTransformForPath(basePath, initialType, expectedType);
+      });
+
+      it('replaces a simple type for a relative path beginning with ".."', () => {
+        const initialType = 'InitialType';
+        const expectedType = 'NonCollisionType';
+        const basePath = '../some/stubbed/path';
+
+        testTypeTransformForPath(basePath, initialType, expectedType);
+      });
+
+      it('replaces a simple type for an absolute path', () => {
+        const initialType = 'InitialType';
+        const expectedType = 'NonCollisionType';
+        const basePath = '~/some/stubbed/path';
+
+        testTypeTransformForPath(basePath, initialType, expectedType);
+      });
+
+      /**
+       * Test helper for performing boilerplate setup to verify some initial type T is transformed to a new type T'.
+       * This helper asserts that the generated type matches T'.
+       *
+       * This helper is meant to be used to test that transformation occurs based on the format/type of `basePath`
+       * provided (e.g. relative vs absolute).
+       *
+       * @param basePath the path to use when running the type resolution
+       * @param initialType the original type found in a class member
+       * @param expectedType the type that is expected to be generated
+       */
+      const testTypeTransformForPath = (basePath: string, initialType: string, expectedType: string): void => {
```

Comment:
sort of a minor thing, but it might be slightly more readable to move this function earlier in the block, up above where it's used, so the reader is encouraged to at least glance at it before reading the tests (I was thinking the same thing for `testTypeIsTransformed` below)

Reply:
I think it could go either way - I generally prefer to order functions in the order they're called, but that's more of a subjective opinion (at least in this scenario in JavaScript where order doesn't affect the execution).  My vote for now would be to leave it as it for now.  In the future I'd like to get ESLint set up in the codebase, and then we can let it warn us when we violate rules (we'll still have to decide/agree on the rules we run though 😉)

Reply:
I did check earlier today out of curiousity and adding eslint w/ basic rules doesn't produce _that_ many errors:

<img width="518" alt="Screen Shot 2022-04-22 at 1 16 33 PM" src="https://user-images.githubusercontent.com/6207644/164763039-e273a2e8-9494-4f6a-874f-745d04b7d4a1.png">

most of them are 'hey you used `any`, don't do that!'